libsolidity/lsp/LanguageServer.cpp

```diff
@@ -50,9 +53,24 @@ using namespace solidity::lsp;
 using namespace solidity::langutil;
 using namespace solidity::frontend;
 
+namespace fs = boost::filesystem;
+
 namespace
 {
 
+bool resolvesToRegularFile(boost::filesystem::path _path)
+{
+	fs::file_status fileStatus = fs::status(_path);
+
+	while (fileStatus.type() == fs::file_type::symlink_file)
+	{
+		_path = boost::filesystem::read_symlink(_path);
+		fileStatus = fs::status(_path);
+	}
```

Comment:
Hmm... I was hoping this would be a simple check but yeah, I guess links can point at other links...

The problem with this is that you can fall into an infinite loop if you have two symlinks pointing at each other. Or a longer cycle.

Now that I think about it, what will the the whole directory scanning do if it finds a symlink loop?